aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SigV4SigningDecorator.kt

```diff
@@ -58,25 +62,49 @@ class SigV4SigningDecorator : RustCodegenDecorator {
         baseCustomizations: List<OperationCustomization>
     ): List<OperationCustomization> {
         return baseCustomizations.letIf(applies(protocolConfig)) {
-            it + SigV4SigningFeature(operation, protocolConfig.runtimeConfig, protocolConfig.serviceShape, protocolConfig.model)
+            it + SigV4SigningFeature(
+                protocolConfig.model,
+                operation,
+                protocolConfig.runtimeConfig,
+                protocolConfig.serviceShape,
+            )
         }
     }
 }
 
-class SigV4SigningConfig(private val sigV4Trait: SigV4Trait) : ConfigCustomization() {
+class SigV4SigningConfig(runtimeConfig: RuntimeConfig, private val sigV4Trait: SigV4Trait) : ConfigCustomization() {
+    private val codegenScope = arrayOf(
+        "SigV4Signer" to RuntimeType(
```

Comment:
todo: pull this in only when event stream signing is needed

Reply:
Fixed.